srtcore/group.cpp

```diff
@@ -3175,19 +3175,72 @@ void CUDTGroup::sendBackup_Buffering(const char* buf, const int len, int32_t& w_
         m_iLastSchedSeqNo = oldest_buffer_seq;
 }
 
+bool CUDTGroup::sendBackup_IsActivationNeeded(const vector<gli_t>&    idlers,
+                                              const vector<gli_t>&    unstable,
+                                              const vector<gli_t>&    sendable,
+                                              const set<uint16_t>     sendable_pri,
+                                              string& activate_reason ATR_UNUSED) const
+{
+    SRT_ASSERT(sendable.size() >= unstable.size());
+    bool need_activate = sendable.size() <= unstable.size(); // <= for sanity, should be just =
+    IF_HEAVY_LOGGING(activate_reason = "BY NO REASON???");
+    if (need_activate)
+    {
+        HLOGC(gslog.Debug,
+              log << "grp/sendBackup: all " << sendable.size() << " links unstable - will activate an idle link");
+        IF_HEAVY_LOGGING(activate_reason = "no stable links");
+    }
+    else if (sendable_pri.empty() ||
+             (!idlers.empty() && FPriorityOrder::check(idlers[0]->weight, *sendable_pri.begin())))
```

Comment:
Likely the base value for sendable_pri to check should be `*--sendable_pri.end()`.